### PR TITLE
Mhv 54563 consolidate api calls rx details

### PR DESCRIPTION
**Pull request by robertroddev**
## Summary

Consolidate API Calls on medicaiton detials view.

Currently the list view contains all the same individual rx data that each medication details view needs. We can store that data when the user clicks on the medication card and use it in the details medication view, instead of having to make another call to the back end.
We will still keep the functionality of calling the backend if the user goes directly to that page or refreshes the page.

## Related issue(s)

[Tech Debt: Consolidate API calls](https://jira.devops.va.gov/browse/MHV-54563)

> User Story: As a Medications developer I want to consolidate the API calls so that we only have to call the API once since it is all the same data

## Testing done

manual testing to ensure the api isn't being called when the user clicks on the medication from the list view, 
unit testing for the redux flow

## Screenshots

no ui was changed

## What areas of the site does it impact?

mhv medications

## Acceptance criteria

- [x]  API calls are consolidated into the least amount possible without effecting data. 
- [ ] 
### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by vmenshutin-bylight — DISMISSED** [review]

**Review by vmenshutin-bylight — DISMISSED** [review]

File: src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js [hunk 1]
```diff
@@ -337,22 +337,21 @@ class MedicationsDetailsPage {
...
+  //     prescriptionDetails,
+  //   ).as('prescriptionDetails');
+  //   cy.get('@prescriptionDetails')
+  //     .its('response')
+  //     .then(res => {
+  //       expect(res.body.data.attributes).to.include({
+  //         cmopNdcNumber: null,
+  //       });
+  //     });
+  // };
```
Comment (micahchiang) [inline src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js:354]: Will this block actually be needed later or can it be removed?

File: src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js [hunk 2]
```diff
@@ -361,22 +360,22 @@ class MedicationsDetailsPage {
...
+  //   //   }`,
+  //   //   prescriptionDetails,
+  //   // ).as('prescriptionDetails');
+  //   cy.get('@prescriptionDetails')
+  //     .its('response')
+  //     .then(res => {
+  //       expect(res.body.data.attributes).to.include({
+  //         prescriptionSource: 'NV',
+  //       });
+  //     });
```
Comment (micahchiang) [inline src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js:377]: Will this block actually be needed later or can it be removed?

File: src/applications/mhv/medications/tests/e2e/pages/MedicationsListPage.js [hunk 3]
```diff
@@ -429,5 +429,39 @@ class MedicationsListPage {
...
   };
+
+  verifyCmopNdcNumberIsNull = () => {
+    // cy.intercept(
+    //   'GET',
+    //   `/my_health/v1/prescriptions/${
+    //     prescriptionDetails.data.attributes.prescriptionId
+    //   }`,
+    //   prescriptionDetails,
+    // ).as('prescriptionDetails');
```
Comment (micahchiang) [inline src/applications/mhv/medications/tests/e2e/pages/MedicationsListPage.js:440]: Will this block actually be needed later or can it be removed?

File: src/applications/mhv/medications/tests/e2e/pages/MedicationsListPage.js [hunk 4]
```diff
@@ -429,5 +429,39 @@ class MedicationsListPage {
...
+  };
+
+  verifyPrescriptionSourceForNonVAMedicationOnDetailsPage = () => {
+    // cy.intercept(
+    //   'GET',
+    //   `/my_health/v1/prescriptions/${
+    //     prescriptionDetails.data.attributes.prescriptionId
+    //   }`,
+    //   prescriptionDetails,
+    // ).as('prescriptionDetails');
```
Comment (micahchiang) [inline src/applications/mhv/medications/tests/e2e/pages/MedicationsListPage.js:457]: Will this block actually be needed later or can it be removed?

File: src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js (hunk 1 above)
Comment (RajiVenkatesh18) [reply src/applications/mhv/medications/tests/e2e/pages/MedicationsDetailsPage.js:354]: I can remove it if this is an issue.

File: src/applications/mhv/medications/tests/e2e/pages/MedicationsListPage.js (hunk 4 above)
Comment (RajiVenkatesh18) [reply src/applications/mhv/medications/tests/e2e/pages/MedicationsListPage.js:457]: yes, I can remove the ones you have mentioned here. Thanks!

**Review by micahchiang — DISMISSED** [review]

**Review by micahchiang — APPROVED** [review]

**Pull request merged**